lib/ayesql/compiler.ex

```diff
@@ -137,21 +137,36 @@ defmodule AyeSQL.Compiler do
   end
 
   @spec create_queries(queries()) :: Macro.t() | [Macro.t()]
-  @spec create_queries(queries(), [Macro.t()]) :: Macro.t() | [Macro.t()]
-  defp create_queries(queries, acc \\ [])
+  @spec create_queries(queries(), [Macro.t()], [queries()]) ::
+          Macro.t() | [Macro.t()]
+  defp create_queries(queries, acc \\ [], metadata \\ [])
 
-  defp create_queries([{nil, nil, fragments}], _acc) do
+  # Handle anonymous queries (3-tuple format)
+  defp create_queries([{nil, nil, fragments}], _acc, _metadata) do
     create_single_query(fragments)
   end
 
-  defp create_queries([], acc) do
-    Enum.reverse(acc)
+  # Handle named queries (4-tuple format with is_fragment)
+  defp create_queries([], acc, metadata) do
+    # Extract fragment names from the original metadata tuples
+    fragment_names = for {name, _docs, _fragments, true} <- metadata, do: name
+
+    fragment_func =
+      quote do
+        def fragment_functions(), do: unquote(fragment_names)
+      end
+
+    [fragment_func | Enum.reverse(acc)]
```

Comment:
we might also want to add
```suggestion
    fragment_names = for {name, _docs, _fragments, true} <- metadata, do: name

    fragment_func =
      quote do
        def fragment_functions(), do: unquote(fragment_names)
      end

    query_names = for {name, _docs, _fragments, false} <- metadata, do: name

    query_func =
      quote do
        def query_functions(), do: unquote(query_names)
      end

    [query_func, fragment_func | Enum.reverse(acc)]
```
so that we have a direct source of all the non-fragment functions also, that might actually be cleaner for my use case because I also have to filter out `__db_runner__`, `__db_options__` and `run` to skip explaining those, and with this I could JUST call explain on each function returned by `query_functions` 🤷‍♂️ just a passing thought, dunno, I don't really need both but might be worth having both.